controllers/replace_misconfigured_pods_test.go

```diff
@@ -387,6 +387,82 @@ var _ = Describe("replace_misconfigured_pods", func() {
 		})
 	})
 
+	When("PVC name and PVC spec doesn't match", func() {
+		It("should need a removal", func() {
+			pvc := corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "Test-storage",
+					Labels: map[string]string{
+						fdbtypes.FDBInstanceIDLabel:   instanceName,
+						fdbtypes.FDBProcessClassLabel: string(fdbtypes.ProcessClassStorage)},
+					Annotations: map[string]string{
+						fdbtypes.LastSpecKey: "f0c8a45ea6c3dd26c2dc2b5f3c699f38d613dab273d0f8a6eae6abd9a9569063",
+					},
+				},
+			}
+			needsRemoval, err := instanceNeedsRemovalForPVC(cluster, pvc)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(needsRemoval).To(BeTrue())
+		})
+	})
+
+	When("PVC name and PVC spec match", func() {
+		It("should not need a removal", func() {
+			pvc := corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "operator-test-1-storage-1337-data",
+					Labels: map[string]string{
+						fdbtypes.FDBInstanceIDLabel:   instanceName,
+						fdbtypes.FDBProcessClassLabel: string(fdbtypes.ProcessClassStorage)},
+					Annotations: map[string]string{
+						fdbtypes.LastSpecKey: "f0c8a45ea6c3dd26c2dc2b5f3c699f38d613dab273d0f8a6eae6abd9a9569063",
+					},
+				},
+			}
+			needsRemoval, err := instanceNeedsRemovalForPVC(cluster, pvc)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(needsRemoval).To(BeFalse())
+		})
+	})
+
+	When("PVC hash doesn't match", func() {
+		It("should need a removal", func() {
+			pvc := corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "operator-test-1-storage-1337-data",
+					Labels: map[string]string{
+						fdbtypes.FDBInstanceIDLabel:   instanceName,
+						fdbtypes.FDBProcessClassLabel: string(fdbtypes.ProcessClassStorage)},
+					Annotations: map[string]string{
+						fdbtypes.LastSpecKey: "1",
+					},
+				},
+			}
+			needsRemoval, err := instanceNeedsRemovalForPVC(cluster, pvc)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(needsRemoval).To(BeTrue())
+		})
+	})
+
+	When("PVC hash and PVC spec hash match", func() {
```

Comment:
This test case is the same like: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/868/files#diff-6c5f00a20e1e8eb48cdb2bb61d6eda7031d06f94211a8d8ab4466748c99c3f15R409-R426 please remove it.